TextStyle/Source/ReferenceWritableKeyPathSetter.swift

```diff
@@ -0,0 +1,24 @@
+struct ReferenceWritableKeyPathSetter<Root> {
+
+    let write: (Root) -> ()
+    
+    private let keyPath: AnyKeyPath
+    private let value: Any
+
+    subscript<Value>(_ keyPath: ReferenceWritableKeyPath<Root, Value>) -> Value? {
+        guard self.keyPath == keyPath else { return nil }
+        return value as? Value
+    }
+
+    init<Value>(keyPath: ReferenceWritableKeyPath<Root, Value>, value: Value) {
+        self.write = { $0[keyPath: keyPath] = value }
+        self.keyPath = keyPath
+        self.value = value
+    }
+}
+
+extension Array {
+    subscript<Root, Value>(_ keyPath: ReferenceWritableKeyPath<Root, Value>) -> Value? where Element == ReferenceWritableKeyPathSetter<Root> {
+        return nil
```

Comment:
How did I leave this unimplemented 🤦

Reply:
Just did: https://github.com/trafi/TextStyle/commit/2c5439363b04aa57586af18a1f0b7a3526fdaf08